pkg/remote/aws/lambda_event_source_mapping_supplier.go

```diff
@@ -0,0 +1,69 @@
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/cloudskiff/driftctl/pkg/remote/aws/repository"
+	remoteerror "github.com/cloudskiff/driftctl/pkg/remote/error"
+
+	"github.com/cloudskiff/driftctl/pkg/remote/deserializer"
+	"github.com/cloudskiff/driftctl/pkg/resource"
+	resourceaws "github.com/cloudskiff/driftctl/pkg/resource/aws"
+	awsdeserializer "github.com/cloudskiff/driftctl/pkg/resource/aws/deserializer"
+	"github.com/cloudskiff/driftctl/pkg/terraform"
+
+	"github.com/aws/aws-sdk-go/service/lambda"
+	"github.com/sirupsen/logrus"
+	"github.com/zclconf/go-cty/cty"
+)
+
+type LambdaEventSourceMappingSupplier struct {
+	reader       terraform.ResourceReader
+	deserializer deserializer.CTYDeserializer
+	client       repository.LambdaRepository
+	runner       *terraform.ParallelResourceReader
+}
+
+func NewLambdaEventSourceMappingSupplier(provider *AWSTerraformProvider) *LambdaEventSourceMappingSupplier {
+	return &LambdaEventSourceMappingSupplier{
+		provider,
+		awsdeserializer.NewLambdaEventSourceMappingDeserializer(),
+		repository.NewLambdaRepository(provider.session),
+		terraform.NewParallelResourceReader(provider.Runner().SubRunner()),
+	}
+}
+
+func (s LambdaEventSourceMappingSupplier) Resources() ([]resource.Resource, error) {
+	functions, err := s.client.ListAllLambdaEventSourceMappings()
+	if err != nil {
+		return nil, remoteerror.NewResourceEnumerationError(err, resourceaws.AwsLambdaEventSourceMappingResourceType)
+	}
+	results := make([]cty.Value, 0)
+	if len(functions) > 0 {
+		for _, function := range functions {
+			fun := *function
+			s.runner.Run(func() (cty.Value, error) {
+				return s.readLambdaEventSourceMapping(fun)
+			})
+		}
+		results, err = s.runner.Wait()
+		if err != nil {
+			return nil, err
+		}
+	}
+	return s.deserializer.Deserialize(results)
+}
+
+func (s LambdaEventSourceMappingSupplier) readLambdaEventSourceMapping(sourceMappingConfig lambda.EventSourceMappingConfiguration) (cty.Value, error) {
+	resFunction, err := s.reader.ReadResource(
+		terraform.ReadResourceArgs{
+			Ty: resourceaws.AwsLambdaEventSourceMappingResourceType,
+			ID: aws.StringValue(sourceMappingConfig.UUID),
```

Comment:
UUID can really be nil there ? I don't think `aws.StringValue` is very usefull there

Reply:
I shouldn't but we have seen strange errors so I think it's a good idea to not crash ?

Reply:
If we want to handle this strange case we should not call `ReadResource` with a nil value IMHO, we should handle this case explicitly before in the loop, and maybe send a `log.Warning` as something went wrong